### PR TITLE
Enable connections for db source/sink

### PR DESCRIPTION
**Pull request by yaojiefeng**

### CODE REVIEW

File: database-plugins/src/main/java/io/cdap/plugin/DBManager.java [hunk 1]
```diff
@@ -53,17 +53,20 @@ public DBManager(ConnectionConfig config) {
       config.connectionArguments), config.jdbcPluginType);
   }
 
-  @Nullable
-  public Class<? extends Driver> validateJDBCPluginPipeline(PipelineConfigurer pipelineConfigurer,
-                                                            String jdbcPluginId, FailureCollector collector) {
+  public void validateCredentials(FailureCollector collector) {
     if (!config.containsMacro(DBConfig.USER) && !config.containsMacro(DBConfig.PASSWORD)
-      && config.getUser() == null && config.getPassword() != null) {
+          && config.getUser() == null && config.getPassword() != null) {
```
Comment (avirmani) [inline database-plugins/src/main/java/io/cdap/plugin/DBManager.java:58]: nit: space?

**Review by avirmani — APPROVED** [review]

File: database-plugins/src/main/java/io/cdap/plugin/db/common/DBBaseConfig.java [hunk 2]
```diff
@@ -144,8 +144,4 @@ public String getReferenceName() {
   public String getColumnNameCase() {
     return columnNameCase;
   }
-
-  public boolean getUseConnection() {
-    return useConnection == null ? false : useConnection;
-  }
```
Comment (sanjanasandeep) [inline database-plugins/src/main/java/io/cdap/plugin/db/common/DBBaseConfig.java:150]: why is this removed?

File: database-plugins/src/main/java/io/cdap/plugin/db/connector/DBConnectorConfig.java [hunk 3]
```diff
@@ -71,6 +71,7 @@
   @Name(JDBC_PLUGIN_NAME)
   @Description("Name of the JDBC plugin to use. This is the value of the 'name' key defined in the JSON file " +
     "for the JDBC plugin.")
```
Comment (sanjanasandeep) [inline database-plugins/src/main/java/io/cdap/plugin/db/connector/DBConnectorConfig.java:73]: Should we have same text as in the doc file? (i.e; add (Macro-enabled))

File: database-plugins/src/main/java/io/cdap/plugin/db/common/DBBaseConfig.java (hunk 2 above)
Comment (yaojiefeng) [reply database-plugins/src/main/java/io/cdap/plugin/db/common/DBBaseConfig.java:150]: Because it was a placeholder to default connection to false, now since we reenable it, we don't need the default value any more

File: database-plugins/src/main/java/io/cdap/plugin/db/connector/DBConnector.java [hunk 4]
```diff
@@ -106,13 +106,9 @@ protected DBConnectorPath getDBConnectorPath(String path) throws IOException {
...
-    Map<String, String> rawProperties = config.getRawProperties().getProperties();
     Map<String, String> properties = new HashMap<>();
-    properties.put(DBConnectorConfig.CONNECTION_STRING, rawProperties.get(DBConnectorConfig.CONNECTION_STRING));
-    properties.put(DBConnectorConfig.JDBC_PLUGIN_NAME, rawProperties.get(DBConnectorConfig.JDBC_PLUGIN_NAME));
-    properties.put(DBConnectorConfig.USER, rawProperties.get(DBConnectorConfig.USER));
-    properties.put(DBConnectorConfig.PASSWORD, rawProperties.get(DBConnectorConfig.PASSWORD));
-    properties.put(DBConnectorConfig.CONNECTION_ARGUMENTS, rawProperties.get(DBConnectorConfig.CONNECTION_ARGUMENTS));
+    properties.put(DBBaseConfig.NAME_USE_CONNECTION, "true");
+    properties.put(DBBaseConfig.NAME_CONNECTION, request.getConnectionWithMacro());
     properties.put(DBBaseConfig.JDBC_PLUGIN_TYPE, DBUtils.PLUGIN_TYPE_JDBC);
```
Comment (sanjanasandeep) [inline database-plugins/src/main/java/io/cdap/plugin/db/connector/DBConnector.java:112]: should there be unit tests for this pr?

File: database-plugins/src/main/java/io/cdap/plugin/db/connector/DBConnectorConfig.java (hunk 3 above)
Comment (yaojiefeng) [reply database-plugins/src/main/java/io/cdap/plugin/db/connector/DBConnectorConfig.java:73]: Already added to source/sink doc, for connector, we don't add those since using macro is rare use case there, except for secure macro

**Review by sanjanasandeep — APPROVED** [review]
LGTM

**Review by seanzhougoogle — APPROVED** [review]

File: database-plugins/src/main/java/io/cdap/plugin/db/connector/DBConnector.java (hunk 4 above)
Comment (yaojiefeng) [reply database-plugins/src/main/java/io/cdap/plugin/db/connector/DBConnector.java:112]: We have ITN for these connectors, but it was not updated yet(it was still based on connection is true), will have a follow up pr to update those

File: database-plugins/src/main/java/io/cdap/plugin/DBManager.java (hunk 1 above)
Comment (yaojiefeng) [reply database-plugins/src/main/java/io/cdap/plugin/DBManager.java:58]: The original indentation was off, since this is separated for a if condition, the indentation should based on the start of the condition
